app/src/views/GrantRegistryNewGrant.vue

```diff
@@ -75,22 +86,30 @@ function useNewGrant() {
   const { poll } = useDataStore();
 
   // Define form fields and parameters
-  const form = ref<{ owner: string; payee: string; metaPtr: string }>({
+  const form = ref<{ owner: string; payee: string; name: string; description: string }>({
     owner: '',
     payee: '',
-    metaPtr: '',
+    name: '',
+    description: '',
   });
   const isFormValid = computed(
-    () => isValidAddress(form.value.owner) && isValidAddress(form.value.payee) && isValidUrl(form.value.metaPtr)
+    () =>
+      isValidAddress(form.value.owner) &&
+      isValidAddress(form.value.payee) &&
+      isDefined(form.value.name) &&
+      isDefined(form.value.description)
   );
 
   /**
    * @notice Creates a new grant, parses logs for the Grant ID, and navigates to that grant's page
    */
   async function createGrant() {
     // Send transaction
-    const { owner, payee, metaPtr } = form.value;
+    const { owner, payee, name, description } = form.value;
     if (!signer.value) throw new Error('Please connect a wallet');
+    const metaPtr = await ipfs
+      .createGrant({ name, description })
+      .then((cid) => ipfs.createMetaPtr({ cid: cid.toString() }));
```

Comment:
Just curious why you use a `.then` instead of
```suggestion
    const cid = await ipfs.createGrant({ name, description })
    const metaPtr = await ipfs.createMetaPtr({ cid: cid.toString() }));
```

Reply:
my general rule is that if something is used once there's no reason to declare it, i.e. since we don't use `cid` elsewhere there's no reason to declare it. i break this sometimes for readability's sake, but `then` clauses are pretty readable to me. Thoughts?